src/test/regress/upgrade/upgrade_test.py

```diff
@@ -0,0 +1,139 @@
+"""upgrade_test
+Usage:
+    upgrade_test --old-bindir=<old-bindir> --new-bindir=<new-bindir> --postgres-srcdir=<postgres-srcdir> [--citus-path=<citus-path>]
+
+Options:
+    --old-bindir=<old-bindir>              The old PostgreSQL executable directory;(ex: '~/.pgenv/pgsql/bin')
+    --new-bindir=<new-bindir>              New postgres binary absolute path(ex: '~/.pgenv/pgsql-11.3/bin')
+    --postgres-srcdir=<postgres-srcdir>    Path to postgres build directory(ex: ~/.pgenv/src/postgresql-11.3/src/test/regress)
+    --citus-path=<citus-path>              Absolute path for citus directory (default: ~/citus)
+"""
+
+from config import *
+from docopt import docopt
+import utils
+import os
+import shutil
+import sys
+
+
+def initialize_temp_dir():
+    if os.path.exists(config[TEMP_DIR]):
+        shutil.rmtree(config[TEMP_DIR])
+    os.mkdir(config[TEMP_DIR])
+    # Give full access to TEMP_DIR so that postgres user can use it.
+    os.chmod(config[TEMP_DIR], 0o777)
+
+
+def initialize_db_for_cluster(pg_path, rel_data_path):
+    utils.run(['mkdir', rel_data_path])
+    for node_name in NODE_NAMES:
+        abs_data_path = os.path.abspath(os.path.join(rel_data_path, node_name))
+        command = [os.path.join(pg_path, 'initdb'),
+                   '--pgdata', abs_data_path, '--username', USER]
+        utils.run(command)
+        add_citus_to_shared_preload_libraries(abs_data_path)
+
+
+def get_add_citus_to_shared_preload_library_cmd(abs_data_path):
+    return 'echo "shared_preload_libraries = \'citus\'" >> {conf_path}'.format(
+        conf_path=os.path.join(abs_data_path, 'postgresql.conf'))
+
+
+def add_citus_to_shared_preload_libraries(abs_data_path):
```

Comment:
given that we now also want to write other (`citus.node_conninfo`) settings before we start postgres it might be nice to make this function a bit more robust.

Let is take a dict of settings to write.
Append it to a file with nice python calls instead of echo :) : https://stackoverflow.com/questions/4706499/how-do-you-append-to-a-file-in-python